.github/workflows/ci-python.yml

```diff
@@ -52,3 +54,25 @@ jobs:
 
       - name: Run tests
         run: poetry run pytest
+
+  docker:
+    name: Build docker image
+    runs-on: ubuntu-latest
+    needs: tests
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 1
+
+      - name: Lint docker image
+        run: make docker-lint
+
+      - name: Build & push docker image
+        run: make docker-login && make docker-build && make docker-push && make docker-sign && make docker-verify
```

Comment:
I'd suggest to split-out the `&& make docker-sign && make docker-verify` steps to a new dedicated step.